### ISSUE TITLE
TouchID on Macbook Pro (M1) support

### ISSUE THREAD

**Issue by yaleman** [kanidm/webauthn-rs#83]
Per the README, I'm reaching out to flag that TouchID on the Macbook's with the touch sensor doesn't seem to work? I'm testing this through [kanidm](https://github.com/kanidm/kanidm) so it could be related to that 😇

**Comment by Firstyear** [comment]
@yaleman When you say 'doesn't work' can you elaborate? Is this in a webbrowser or the kanidm cli?

**Comment by yaleman** [comment]
Sorry this is in the kanidm cli, I can't successfully webauthn register, it says it'll flash but ... nothing happens. Touching the sensor which would normally trigger it doesn't make the process continue.

```
$ cargo run --bin kanidm -- account credential register_webauthn yaleman ap-touchid --debug verbose
    Finished dev [unoptimized + debuginfo] target(s) in 0.30s
     Running `target/debug/kanidm account credential register_webauthn yaleman ap-touchid --debug verbose`
[2021-07-27T02:53:46Z DEBUG kanidm_client] Attempting to load configuration from "/etc/kanidm/config"
[2021-07-27T02:53:46Z DEBUG kanidm_client] Configuration file "/etc/kanidm/config" not found, skipping.
[2021-07-27T02:53:46Z DEBUG kanidm_client] Attempting to load configuration from "/Users/yaleman/.config/kanidm"
[2021-07-27T02:53:46Z DEBUG kanidm_client] Successfully opened configuration file "/Users/yaleman/.config/kanidm"
[2021-07-27T02:53:46Z DEBUG kanidm_cli::common] Successfully read configuration from /Users/yaleman/.config/kanidm
[2021-07-27T02:53:46Z DEBUG kanidm_cli::session] Attempting to read tokens from "/Users/yaleman/.cache/kanidm_tokens"
Multiple authentication tokens exist. Please select one: yaleman
[2021-07-27T02:53:48Z DEBUG kanidm_cli::common] Index of the chosen menu item: 1
[2021-07-27T02:53:48Z INFO  kanidm_cli::common] Using cached token for name yaleman
[2021-07-27T02:53:48Z DEBUG kanidm_client::asynchronous] opid -> "f2b53feb-76a9-4d0b-9a37-38b11b9d9e9e"
Your authenticator will now flash for you to interact with.
[2021-07-27T02:53:48Z DEBUG webauthn_authenticator_rs] effective domain -> "kanidm.example.com"
[2021-07-27T02:53:48Z DEBUG webauthn_authenticator_rs] relying party id -> "kanidm.example.com"
[2021-07-27T02:53:48Z DEBUG webauthn_authenticator_rs] Found -> [("public-key", fffffffffffffff9), ("public-key", fffffffffffffeff)]
[2021-07-27T02:53:48Z DEBUG webauthn_authenticator_rs] client_data_json -> "{\"type\":\"webauthn.create\",\"challenge\":\"MIVjVJPUaD3SI4E5n-r4GLC2YgcuuD2QMxitk00felA\",\"origin\":\"https://kanidm.example.com\",\"tokenBinding\":null}"
[2021-07-27T02:53:48Z DEBUG webauthn_authenticator_rs] client_data_json_hash -> [db, 36, 1c, 24, d2, b9, 65, 4c, 1d, e2, df, 16, da, a, ac, 21, 96, 23, f2, 10, 91, 87, e2, f0, 0, 52, fa, 2c, 29, ac, f4, 29]
[2021-07-27T02:53:48Z DEBUG webauthn_authenticator_rs::u2fhid] flags -> (empty)
```

**Comment by Firstyear** [comment]
Yeah this is a bug in webauthn-authenticator-rs. At the moment only ctap1 tokens are supported, as touchid works differently. I've been talking with the new owner of authenticator-rs which is the underlying library. They are currently adding ctap2 support (for u2f with pin) and we want to add touchid later as well.

**Comment by Firstyear** [comment]
tl;dr - on the cli, it's yubikey only atm.

**Comment by yaleman** [comment]
Ok, thanks 😄

**Comment by Firstyear** [comment]
even that milestone won't enable touchid support, it's a different interface all together I think.

**Comment by Firstyear** [comment]
https://github.com/kanidm/webauthn-authenticator-rs/issues/3
